app/models/manageiq/providers/hawkular/inventory/operation_notification.rb

```diff
@@ -33,13 +33,15 @@ def emit
               unless mw_entity
                 EmsEvent.add_queue(
                   'add', manager.id,
-                  :ems_id          => manager.id,
-                  :source          => 'HAWKULAR',
-                  :timestamp       => Time.zone.now,
-                  :event_type      => args.event_type(mw_server),
-                  :message         => args.event_message(mw_server),
-                  :middleware_ref  => mw_server.ems_ref,
-                  :middleware_type => mw_server.class.name.demodulize
+                  :ems_id                 => manager.id,
+                  :source                 => 'HAWKULAR',
+                  :timestamp              => Time.zone.now,
+                  :event_type             => args.event_type(mw_server),
+                  :message                => args.event_message(mw_server),
+                  :middleware_server_id   => mw_server.id,
+                  :middleware_server_name => mw_server.name,
```

Comment:
wasn't the original request for the hostname  to be added? (which is not the server name, it's another property), or it's converted somehow on the way?

Reply:
I asked Jdoyle in Kira about because I thought it was the hostname (I can do it too without link to the host inventory) but it's make sense to be the middleware server.

Reply:
These two fields, `middleware_server_id` and `middleware_server_name` were previously being set in [EmsEvent class](https://github.com/ManageIQ/manageiq/blob/master/app/models/ems_event.rb#L146-L155). Indeed, I remember the server was correctly being linked in the events.
Perhaps, this is an issue injected by #60 and may be better to fix `EmsEvent` for a broader scope.

Reply:
Mmm after https://github.com/ManageIQ/manageiq-ui-classic/pull/2958 we can see the middleware server in user operations, maybe this is different @israel-hdez

Reply:
ManageIQ/manageiq-ui-classic#2958 is doing the same: it's setting manually those fields. So, if you fix `EmsEvent` you'll be fixing both the UI part and the events in this PR... And any other similar events.

Reply:
(Try reverting #60 in your env and, most likely, you will see the links to the server without the need of these PRs)

Reply:
The only thing I see is  inventory_object.assign_attributes maybe this is the different, ok I am gonna check this. thanks